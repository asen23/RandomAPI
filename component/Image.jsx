export default function Image({ src, alt }) {
    return (
        // eslint-disable-next-line @next/next/no-img-element
        <img src={src} alt={alt} className="max-w-[80%] max-h-[60vh]" />
    )
}