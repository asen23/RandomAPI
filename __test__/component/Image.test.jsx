import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Image from "component/Image"

const dummyImg = {
    src: "http://example.com/tes.png",
    alt: "an alternate text",
}

describe("Image", () => {
    it("renders unchanged", async () => {
        const { asFragment } = render(
            <Image src={dummyImg.src} alt={dummyImg.alt} />
        )
        expect(asFragment()).toMatchSnapshot()
    })

    it("renders img correctly", () => {
        render(<Image src={dummyImg.src} alt={dummyImg.alt} />)

        const image = screen.getByAltText(dummyImg.alt)
        expect(image.src).toContain(dummyImg.src)
    })
})
