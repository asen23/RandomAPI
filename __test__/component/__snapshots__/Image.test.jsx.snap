// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders unchanged 1`] = `
<DocumentFragment>
  <img
    alt="an alternate text"
    class="max-w-[80%] max-h-[60vh]"
    src="http://example.com/tes.png"
  />
</DocumentFragment>
`;
